src/pages/index.js

```diff
@@ -41,70 +55,243 @@ addFormValidator.enableValidation();
+//////////////////////////////////////////////////////////////////////////
 
-const addCardModal = new PopupWithForm({
-  popupSelector: "#add-card-modal",
-  handleFormSubmit: (data) => {
-    const cardElement = createCard({ name: data.title, link: data.url });
-    section.addItem(cardElement);
-    cardForm.reset();
-    toggleAddButtonState();
+// Initialize API //
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c58369c4-dcd4-4208-8726-c10df21880b5",
+    "Content-Type": "application/json",
   },
 });
 
-// Initialize User Info
+// Initialize User Info //
 const userInfo = new UserInfo({
   nameSelector: ".profile__name",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__avatar",
 });
 
-// Intialize Section
+// Intialize Section //
 const section = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: (cardData) => {
-      const cardElement = createCard(cardData);
-      section.addItem(cardElement);
+      section.addItem(cardData);
     },
   },
   ".cards"
 );
 
-section.renderItems();
+console.log("Section instance:", section);
 
-// Modal Event Listeners
-editProfileModal.setEventListeners();
-addCardModal.setEventListeners();
-imagePreviewModal.setEventListeners();
+// Fetch and Render Initial Data//
+api
+  .getAllData()
+  .then(({ user, cards }) => {
+    console.log("Initial user data:", user);
+    console.log("Initial cards data:", cards);
+    const cardElement = cards.map((cardData) => createCard(cardData));
+    section.renderItems(cardElement);
+    userInfo.setUserInfo(user);
+    // if (cards.length === 0) {
+    //   console.log("No cards found on the server. Adding initial cards.");
+    //   addInitialCards(initialCards)
+    //     .then(() => api.getInitialCards())
+    //     .then((newCards) => {
+    //       console.log("New cards added from initial cards:", newCards);
+    //       section.renderItems(newCards);
+    //     });
+    // } else {
+    //   console.log("Rendering cards from the server:", cards);
+    //   section.renderItems(items);
+    // }
+  })
+  .catch((error) => console.error("Error loading data", error));
+
+const editAvatarModal = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  handleFormSubmit: (data) => {
+    const userAvatar = {
+      avatar: data.avatar,
+    };
+    api
+      .updateUserAvatar(userAvatar)
+      .then((updatedUserInfo) => {
+        userInfo.setUserInfo(updatedUserInfo);
+        editAvatarModal.close();
+      })
+      .catch((error) => console.error("Error updating user avatar:", error));
+  },
+});
+
+const editProfileModal = new PopupWithForm({
+  popupSelector: "#edit-profile-modal",
+  // handleFormSubmit: (data) => {
+  //   userInfo.setUserInfo({ name: data.name, job: data.job });
+  // },
+  handleFormSubmit: (data) => {
+    const userData = {
+      name: data.name,
+      about: data.about,
+    };
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```